assets/js/src/segments/dynamic/form.tsx

```diff
@@ -64,6 +70,9 @@ export const Form: React.FunctionComponent<Props> = ({
               {MailPoet.I18n.t('name')}
             </label>
           </Heading>
+          {(errors.length > 0 && (
+            <APIErrorsNotice errors={errors.map((error) => ({ message: error }))} />
+          ))}
```

Comment:
I think that the APIErrorsNotice may actually display more types of errors (e.g. non unique form name) so we shouldn't display all within the form name input. I suggest moving those error notices at the very beginning of the form.